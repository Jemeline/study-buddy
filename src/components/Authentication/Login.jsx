import React, { useState } from "react";
import { Container, Form, Button } from "react-bootstrap";
import { Redirect } from "react-router-dom";
import { withFirebase } from "../../utils/Firebase.js";

const Login = ({ Firebase }) => {
    const [loggedIn, setLoggedIn] = useState(false);
    const handleLogin = async e => {
        e.preventDefault();
        const { email, password } = e.target.elements;
        try {
            const user = await Firebase.login(email.value, password.value);
            console.log(user);
            setLoggedIn(true);
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <Container className="loginContainer">
            {loggedIn ? <Redirect to="/student/dashboard" /> : null}
            <Form className="loginForm" onSubmit={handleLogin}>
                <h1>Firebase Login</h1>

                <Form.Control
                    name="email"
                    type="email"
                    placeholder="Email Address"
                />
                <Form.Control
                    name="password"
                    type="password"
                    placeholder="Password"
                />
                <Button type="submit">Login</Button>
            </Form>
        </Container>
    );
};

export default withFirebase(Login);
