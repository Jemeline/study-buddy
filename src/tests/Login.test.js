import { render, screen, cleanup, fireEvent,waitFor } from '@testing-library/react';
import Login from '../components/Authentication/Login.component';


function setIsLoggedIn() {};
function setTab() {};
function setFirst() {};
function setId() {};
function setEmail() {};

test('renders Login component', () => {
  render(<Login />);
  const element = screen.getByTestId("Login");
  expect(element).toBeInTheDocument();
});
 
// it('should take a snapshot of Login', () => {
//     const { asFragment } = render(<Login />)
//     expect(asFragment(<Login />)).toMatchSnapshot()
// });

it('Login form is rendered correctly', () => {
    render(<Login />);
    expect(screen.getByTestId('login-button')).toBeInTheDocument();
    expect(screen.getByTestId('login-email-input')).toBeInTheDocument();
    expect(screen.getByTestId('login-password-input')).toBeInTheDocument();
});

it('login button should be disabled initially', () => {
    render(<Login />); 
    expect(screen.getByTestId('login-button')).toBeDisabled()
});

it("login button should be enabled when valid email entered in input box", () => {
    render(<Login />);
    const button = screen.getByTestId('login-button');
    const emailInput = screen.getByTestId('login-email-input');
    fireEvent.change(emailInput, { target: { value: "rtl@unittesting.com" } });
    expect(button).toBeEnabled();
});

it("login button should be enabled when invlalid email entered in input box", () => {
    render(<Login />);
    const button = screen.getByTestId('login-button');
    const emailInput = screen.getByTestId('login-email-input');
    fireEvent.change(emailInput, { target: { value: "rtl" } });
    expect(button).toBeDisabled();
});

it("alert should appear when incomplete credentials entered", () => {
    render(<Login />);
    const button = screen.getByTestId('login-button');
    const emailInput = screen.getByTestId('login-email-input');
    fireEvent.change(emailInput, { target: { value: "rtl@unittesting.com" } });
    expect(button).toBeEnabled();
    fireEvent.click(button);
    expect(screen.getByTestId('login-alert-incomplete-creds')).toBeInTheDocument();
});

it("login should fail when invalid credentials entered", async () => {
    render(<Login />);
    const button = screen.getByTestId('login-button');
    const emailInput = screen.getByTestId('login-email-input');
    const passwordInput = screen.getByTestId('login-password-input');
    fireEvent.change(emailInput, { target: { value: "rtl@unittesting.com" } });
    fireEvent.change(passwordInput, { target: { value: "test" } });
    expect(button).toBeEnabled();
    fireEvent.click(button);
});

it("login should succeed when valid credentials entered", async () => {
    render(<Login setIsLoggedIn={setIsLoggedIn} setTab={setTab} setFirst={setFirst} setEmail={setEmail} setId={setId}/>);
    const button = screen.getByTestId('login-button');
    const emailInput = screen.getByTestId('login-email-input');
    const passwordInput = screen.getByTestId('login-password-input');
    fireEvent.change(emailInput, { target: { value: "rtl@unittesting.com" } });
    fireEvent.change(passwordInput, { target: { value: "TestTest1!" } });
    expect(button).toBeEnabled();
    fireEvent.click(button);
});