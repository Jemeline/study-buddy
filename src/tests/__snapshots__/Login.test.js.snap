// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of Login 1`] = `
<DocumentFragment>
  <div
    class="Login"
    data-testid="Login"
  >
    <div
      class="container"
      style="width: 35vw; margin: auto;"
    >
      <div
        class="col"
      >
        <img
          src="study-buddy-tagline.png"
          style="height: 35vh;"
        />
        <br />
        <form
          class="form"
        >
          <div
            class="form-group"
          >
            <input
              aria-invalid="false"
              class="form-control"
              name="Email"
              placeholder="Email"
              style="border-radius: 14px;"
              type="email"
            />
            <div
              class="invalid-feedback"
            >
              There is an issue with your email. Please input a correct email.
            </div>
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="passwordLogin"
              placeholder="Password"
              style="border-radius: 14px;"
              type="password"
            />
          </div>
        </form>
        <button
          class="btn btn-secondary btn-lg disabled"
          data-testid="login-button"
          disabled=""
          style="background-color: rgb(39, 38, 92); color: rgb(255, 255, 255); width: 50%; border-radius: 14px;"
          type="submit"
        >
           Sign In
        </button>
        <br />
        <div
          style="display: flex; justify-content: center;"
        >
          <div
            hidden=""
            style="fill: #27265c; height: 10%; width: 10%;"
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
  
              <path
                d="M0 12 V20 H4 V12z"
                transform="translate(0 0)"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  begin="0"
                  calcMode="spline"
                  dur="1.5s"
                  keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
                  keyTimes="0;0.3;0.6;1"
                  repeatCount="indefinite"
                  type="translate"
                  values="0 0; 28 0; 0 0; 0 0"
                />
                
  
              </path>
              
  
              <path
                d="M0 12 V20 H4 V12z"
                opacity="0.5"
                transform="translate(0 0)"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  begin="0.1s"
                  calcMode="spline"
                  dur="1.5s"
                  keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
                  keyTimes="0;0.3;0.6;1"
                  repeatCount="indefinite"
                  type="translate"
                  values="0 0; 28 0; 0 0; 0 0"
                />
                
  
              </path>
              
  
              <path
                d="M0 12 V20 H4 V12z"
                opacity="0.25"
                transform="translate(0 0)"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  begin="0.2s"
                  calcMode="spline"
                  dur="1.5s"
                  keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
                  keyTimes="0;0.3;0.6;1"
                  repeatCount="indefinite"
                  type="translate"
                  values="0 0; 28 0; 0 0; 0 0"
                />
                
  
              </path>
              

            </svg>
            

          </div>
        </div>
        <br />
        Don't Have an Account?
        <button
          class="btn btn-link"
          type="button"
        >
           Sign Up Now
        </button>
        <br />
        <button
          class="btn btn-link"
          type="button"
        >
          Forgot Password?
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
